conf/nginx.conf

```
@@ -12,6 +12,9 @@ location ^~ /.well-known {
 
   location = /.well-known/carddav     { return 301 __PATH__/remote.php/dav/; }
   location = /.well-known/caldav      { return 301 __PATH__/remote.php/dav/; }
+ 
+  # Anything else is dynamically handled by Nextcloud
+  location ^~ /.well-known          { return 301 /nextcloud/index.php$uri; }
```

```suggestion
  location ^~ /.well-known          { return 301 __PATH__/index.php$uri; }
```

But, the problem with this line is if another app which uses `.well-known/something` is installed on the same domain, then nextcloud will handle it instead of this app

---

Okay, I will look further and see if I find something (I know nothing about all this, so don’t expect a lot form me 😄)

---

A workaround we found is the lines above: https://github.com/YunoHost-Apps/nextcloud_ynh/blob/ee18542f4bc7dd888e7f8bfbae92d9e076c27ccb/conf/nginx.conf#L6-L11
This requires to edit the file manually for users, which is not really YunoHost-friendly

---

And adding only 
```
 location = /.well-known/webfinger   { return 301  __PATH__/index.php$uri; }
 location = /.well-known/nodeinfo   { return 301 __PATH__/index.php$uri; }
```

is not a good alternative as well ?

---

Yes probably :+1: 

After some tests, I can confirm it works. Let's merge it and I'll patch nginx conf